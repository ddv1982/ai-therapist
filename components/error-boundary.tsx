'use client';

import React, { Component, ReactNode } from 'react';
import { logger } from '@/lib/logger';

interface ErrorBoundaryState {
  hasError: boolean;
  error?: Error;
  errorInfo?: string;
}

interface ErrorBoundaryProps {
  children: ReactNode;
  fallback?: ReactNode;
  onError?: (error: Error, errorInfo: string) => void;
}

export class ErrorBoundary extends Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error: Error): ErrorBoundaryState {
    // Update state so the next render will show the fallback UI
    return {
      hasError: true,
      error,
    };
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    // Log the error with structured logging
    logger.error('React component error caught by boundary', {
      component: this.constructor.name,
      stack: error.stack,
      componentStack: errorInfo.componentStack,
      errorBoundary: true
    }, error);

    // Call optional error handler
    if (this.props.onError) {
      this.props.onError(error, errorInfo.componentStack);
    }

    this.setState({
      errorInfo: errorInfo.componentStack,
    });
  }

  private handleRetry = () => {
    this.setState({ hasError: false, error: undefined, errorInfo: undefined });
  };

  private handleReload = () => {
    window.location.reload();
  };

  render() {
    if (this.state.hasError) {
      // Custom fallback UI
      if (this.props.fallback) {
        return this.props.fallback;
      }

      // Default error UI
      return (
        <div className="min-h-screen bg-background flex items-center justify-center p-4">
          <div className="max-w-md w-full bg-card rounded-lg border p-6 text-center">
            <div className="mb-4">
              <div className="w-12 h-12 bg-destructive/10 rounded-full flex items-center justify-center mx-auto mb-4">
                <svg
                  className="w-6 h-6 text-destructive"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth={2}
                    d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-2.5L13.732 4c-.77-.833-1.964-.833-2.732 0L3.732 16c-.77.833.192 2.5 1.732 2.5z"
                  />
                </svg>
              </div>
              <h2 className="text-xl font-semibold text-foreground mb-2">
                Something went wrong
              </h2>
              <p className="text-muted-foreground text-sm mb-6">
                We're sorry, but there was an unexpected error. Your conversation data is safe.
              </p>
            </div>

            <div className="space-y-3">
              <button
                onClick={this.handleRetry}
                className="w-full bg-primary text-primary-foreground hover:bg-primary/90 px-4 py-2 rounded-md text-sm font-medium transition-colors"
              >
                Try Again
              </button>
              <button
                onClick={this.handleReload}
                className="w-full bg-secondary text-secondary-foreground hover:bg-secondary/80 px-4 py-2 rounded-md text-sm font-medium transition-colors"
              >
                Reload Page
              </button>
            </div>

            {process.env.NODE_ENV === 'development' && this.state.error && (
              <details className="mt-6 text-left">
                <summary className="cursor-pointer text-sm text-muted-foreground mb-2">
                  Error Details (Development Only)
                </summary>
                <pre className="text-xs bg-muted p-3 rounded overflow-auto max-h-32">
                  {this.state.error.toString()}
                  {this.state.errorInfo}
                </pre>
              </details>
            )}
          </div>
        </div>
      );
    }

    return this.props.children;
  }
}

// Specialized error boundaries for different parts of the app

export function ChatErrorBoundary({ children }: { children: ReactNode }) {
  return (
    <ErrorBoundary
      fallback={
        <div className="flex-1 flex items-center justify-center p-4">
          <div className="text-center">
            <div className="w-12 h-12 bg-destructive/10 rounded-full flex items-center justify-center mx-auto mb-4">
              <svg className="w-6 h-6 text-destructive" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M8 12h.01M12 12h.01M16 12h.01M21 12c0 4.418-3.582 8-8 8a8.955 8.955 0 01-4.906-1.469L3 21l2.469-5.094A8.955 8.955 0 013 12c0-4.418 3.582-8 8-8s8 3.582 8 8z" />
              </svg>
            </div>
            <h3 className="text-lg font-semibold mb-2">Chat Error</h3>
            <p className="text-muted-foreground text-sm mb-4">
              There was an issue with the chat interface. Please refresh to continue.
            </p>
            <button
              onClick={() => window.location.reload()}
              className="bg-primary text-primary-foreground px-4 py-2 rounded-md text-sm"
            >
              Refresh Chat
            </button>
          </div>
        </div>
      }
      onError={(error) => {
        logger.error('Chat component error', {
          component: 'ChatErrorBoundary',
          userAction: 'chat_interaction'
        }, error);
      }}
    >
      {children}
    </ErrorBoundary>
  );
}

export function SettingsErrorBoundary({ children }: { children: ReactNode }) {
  return (
    <ErrorBoundary
      fallback={
        <div className="p-4 bg-card rounded-lg border">
          <div className="text-center">
            <h3 className="text-sm font-medium mb-2">Settings Error</h3>
            <p className="text-xs text-muted-foreground mb-3">
              Unable to load settings. Using defaults.
            </p>
            <button
              onClick={() => window.location.reload()}
              className="text-xs bg-secondary px-3 py-1 rounded"
            >
              Reload
            </button>
          </div>
        </div>
      }
      onError={(error) => {
        logger.error('Settings component error', {
          component: 'SettingsErrorBoundary',
          userAction: 'settings_interaction'
        }, error);
      }}
    >
      {children}
    </ErrorBoundary>
  );
}