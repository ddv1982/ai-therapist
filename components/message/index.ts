/**
 * Message Component System - Clean Public API
 */

// Main components
export { Message } from './Message';
export { MessageAvatar } from './MessageAvatar';
export { MessageContent } from './MessageContent';
export { MessageTimestamp } from './MessageTimestamp';

// Types
export type { MessageData, MessageProps } from './Message';

// Re-export design system types for convenience
export type { MessageRole } from '@/lib/design-system/message';