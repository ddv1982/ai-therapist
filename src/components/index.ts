/**
 * Components Domain - Master Exports
 * Centralized exports for all component domains
 */

// Domain-specific exports
export * as auth from './auth';
export * as chat from './chat';
export * as therapy from './therapy';
export * as messages from './messages';
export * as ui from './ui';
export * as providers from './providers';

// Direct exports for commonly used components
export * from './messages';
export * from './providers';

// UI components (flattened for convenience)
export * from './ui/primitives';
export * from './ui/enhanced';
export * from './ui/layout';

// Backward compatible exports for easier migration
// Auth components
export * from './auth/auth-guard';
export * from './auth/security-settings';

// Chat components
export * from './chat/chat-interface';
export * from './chat/chat-settings';
export * from './chat/session-controls';
export * from './chat/session-sidebar';

// Message components
export * from './messages/message';
export * from './messages/message-avatar';
export * from './messages/message-content';
export * from './messages/message-timestamp';

// Provider components
export * from './providers/theme-provider';

// Layout exports for backward compatibility
export { ErrorBoundary } from './ui/layout/error-boundary';