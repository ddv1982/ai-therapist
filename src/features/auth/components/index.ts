/**
 * Authentication Components Domain
 * Components for authentication and user management
 */

export * from './auth-guard';