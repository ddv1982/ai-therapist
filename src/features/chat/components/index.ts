/**
 * Chat Components Domain
 * Components for chat interface, sessions, and messaging functionality
 */

export * from './chat-settings';
export * from './session-controls';
export * from './session-sidebar';
export * from './typing-indicator';
export * from './virtualized-message-list';