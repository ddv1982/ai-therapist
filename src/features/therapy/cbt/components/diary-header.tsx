import React from 'react';
import { Button } from '@/components/ui/button';
import { ArrowLeft, Brain } from 'lucide-react';
import { cn } from '@/lib/utils/utils';
import {useTranslations} from 'next-intl';
import { getStepInfo } from '@/features/therapy/cbt/utils/step-mapping';
import type { CBTStep } from '@/features/therapy/cbt/hooks/use-cbt-chat-experience';

interface DiaryHeaderProps {
  isMobile: boolean;
  isCBTActive: boolean;
  cbtCurrentStep: CBTStep;
  onBack: () => void;
}

export function DiaryHeader({ isMobile, isCBTActive, cbtCurrentStep, onBack }: DiaryHeaderProps) {
  const t = useTranslations('cbt');

  if (isMobile) {
    return (
      <div className="border-b bg-card/70 backdrop-blur-md sticky top-0 z-10">
        <div className="max-w-4xl mx-auto px-3 py-2">
          <div className="flex items-center justify-between gap-2">
            <Button
              variant="ghost"
              size="sm"
              onClick={onBack}
              className={cn("flex items-center gap-1 h-8 px-2")}
            >
              <ArrowLeft className="w-4 h-4" />
              <span className="text-sm">{t('nav.back')}</span>
            </Button>
            {isCBTActive && cbtCurrentStep !== 'complete' && (
              <span className="text-xs text-primary">
                Step {getStepInfo(cbtCurrentStep).stepNumber}/{getStepInfo(cbtCurrentStep).totalSteps}
              </span>
            )}
          </div>
          {isCBTActive && cbtCurrentStep !== 'complete' && (
            <div className="w-full bg-muted rounded-full h-1 mt-2">
              <div 
                className="bg-primary h-1 rounded-full transition-all duration-500 ease-out"
                style={{ width: `${(getStepInfo(cbtCurrentStep).stepNumber / getStepInfo(cbtCurrentStep).totalSteps) * 100}%` }}
              ></div>
            </div>
          )}
        </div>
      </div>
    );
  }

  return (
    <div className="border-b bg-card/50 backdrop-blur-md sticky top-0 z-10">
      <div className="max-w-4xl mx-auto px-4 sm:px-6 py-4">
        <div className="flex items-center justify-between">
          <div className="flex items-center gap-4">
            <Button
              variant="ghost"
              size="sm"
              onClick={onBack}
              className="flex items-center gap-2 hover:bg-accent hover:text-accent-foreground"
            >
              <ArrowLeft className="w-4 h-4" />
              <span className="hidden sm:inline">{t('nav.backToChat')}</span>
            </Button>
            <div className="flex items-center gap-3">
              <div className="h-12 w-12 rounded-xl bg-primary flex items-center justify-center shadow-lg">
                <Brain className="w-6 h-6 text-primary-foreground" />
              </div>
              <div>
                <h1 className="text-3xl font-semibold">
                  Interactive CBT Session
                </h1>
                {isCBTActive && cbtCurrentStep !== 'complete' && (
                  <p className="text-sm text-muted-foreground capitalize mt-1">
                    {cbtCurrentStep.replace('-', ' ')}
                  </p>
                )}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}

export default DiaryHeader;
