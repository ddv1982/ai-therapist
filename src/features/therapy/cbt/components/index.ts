export { SituationSection } from './situation-section';
export { EmotionsSection } from './emotions-section';
export { ThoughtsSection } from './thoughts-section';
export { SchemaSection } from './schema-section';
export { ChallengeSection } from './challenge-section';
export { ResultsSection } from './results-section';
export { EmotionScale } from './emotion-scale';
export { ArrayField } from './array-field';