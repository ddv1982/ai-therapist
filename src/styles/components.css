@layer components {
  /* Spacing utilities */
  .spacing-therapy-xs { @apply p-therapy-xs; }
  .spacing-therapy-sm { @apply p-therapy-sm; }
  .spacing-therapy-md { @apply p-therapy-md; }
  .spacing-therapy-lg { @apply p-therapy-lg; }
  .spacing-therapy-xl { @apply p-therapy-xl; }
  .spacing-therapy-2xl { @apply p-therapy-2xl; }

  /* Color variants */
  .therapy-primary { @apply bg-primary text-primary-foreground shadow-sm; }
  .therapy-secondary { @apply bg-secondary text-secondary-foreground; }
  .therapy-accent { @apply bg-accent text-accent-foreground shadow-sm; }
  .therapy-muted { @apply bg-muted text-muted-foreground; }
  .therapy-card { @apply bg-card text-card-foreground border border-border rounded-therapy-md shadow-sm; }

  /* Buttons */
  .therapy-button-primary { 
    @apply inline-flex items-center justify-center px-therapy-md py-therapy-xs rounded-therapy-sm text-base font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-primary text-primary-foreground shadow-sm hover:bg-primary/90 active:scale-95; 
  }
  .therapy-button-secondary { 
    @apply inline-flex items-center justify-center px-therapy-md py-therapy-xs rounded-therapy-sm text-base font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-secondary text-secondary-foreground hover:bg-secondary/80 active:scale-95; 
  }
  .therapy-button-accent { 
    @apply inline-flex items-center justify-center px-therapy-md py-therapy-xs rounded-therapy-sm text-base font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-accent text-accent-foreground shadow-sm hover:bg-accent/90 active:scale-95; 
  }
  .therapy-button-ghost { 
    @apply inline-flex items-center justify-center px-therapy-md py-therapy-xs rounded-therapy-sm text-base font-medium transition-all duration-200 focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none hover:bg-accent/10 hover:text-accent-foreground active:scale-95; 
  }

  /* Forms */
  .therapy-form-group { @apply space-y-therapy-xs; }
  .therapy-form-label { @apply text-sm font-medium text-foreground; }
  .therapy-form-description { @apply text-xs text-muted-foreground; }
  .therapy-form-error { @apply text-xs text-destructive; }
  .therapy-input { @apply w-full px-therapy-sm py-therapy-xs bg-background border border-input rounded-therapy-sm text-base focus:outline-none focus:ring-2 focus:ring-ring focus:border-transparent placeholder:text-muted-foreground disabled:opacity-50 disabled:cursor-not-allowed transition-colors duration-200; }

  /* Modals */
  .therapy-modal { @apply bg-background border border-border rounded-therapy-lg shadow-xl backdrop-blur-sm; }
  .therapy-modal-content { @apply p-therapy-md max-h-[90vh] overflow-y-auto; }
  .therapy-modal-header { @apply p-therapy-md border-b border-border flex items-center justify-between; }
  .therapy-modal-footer { @apply p-therapy-md border-t border-border flex items-center justify-end gap-therapy-sm; }

  /* Chat-specific */
  .message-spacing { @apply mb-4; }
  .message-padding { @apply p-2; }
  .message-content-user, .message-content-assistant { @apply leading-relaxed; }

  /* Glassmorphism */
  .glass-card { @apply bg-card/80 backdrop-blur-md border border-border/50 shadow-lg; }

  /* Animations */
  .animate-message-in { animation: fade-in 0.5s ease-out; }
  .animate-slide-in { animation: slide-in 0.3s ease-out; }

  /* Scrollbars */
  .custom-scrollbar::-webkit-scrollbar { width: 0; height: 0; }
  .custom-scrollbar { scrollbar-width: none; }
  .custom-scrollbar::-webkit-scrollbar-thumb { @apply bg-muted-foreground/20 rounded-full; }

  /* Tables */
  .therapeutic-table { @apply w-full border-primary/20 bg-primary/5 rounded-lg overflow-hidden; table-layout: fixed; }
  .table-container { @apply rounded-md border overflow-hidden mb-4; }
  .table-system { @apply w-full border-collapse; }
  .table-header-therapeutic { @apply bg-primary/10 text-primary-foreground/90 font-semibold py-3 px-4; }
  .table-row-therapeutic { @apply hover:bg-primary/10 transition-colors border-b border-primary/10; }
  .table-cell-therapeutic { @apply py-3 px-4 text-left; }

  /* Cards */
  .therapeutic-card { @apply transition-all duration-200 ease-out relative overflow-hidden; }
  .therapeutic-card:hover { @apply -translate-y-1; }
  .therapeutic-card::before { @apply absolute inset-0 bg-gradient-to-br from-primary/5 via-transparent to-accent/5 opacity-0 transition-opacity duration-300 pointer-events-none; content: ''; }
  .therapeutic-card:hover::before { @apply opacity-100; }

  /* Sliders */
  input[type="range"] { -webkit-appearance: none; appearance: none; background: transparent; cursor: pointer; }
  input[type="range"]::-webkit-slider-track { @apply bg-muted h-2 rounded-lg; }
  input[type="range"]::-webkit-slider-thumb { -webkit-appearance: none; @apply w-5 h-5 rounded-full bg-primary shadow-lg cursor-pointer transition-all; }
  input[type="range"]::-webkit-slider-thumb:hover { @apply scale-110 shadow-xl; }
}
