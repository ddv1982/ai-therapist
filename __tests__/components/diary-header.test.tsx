import React from 'react';
import { render } from '@testing-library/react';
import { NextIntlClientProvider } from 'next-intl';
import { DiaryHeader } from '@/features/therapy/cbt/components/diary-header';
import enMessages from '@/i18n/messages/en.json';

// Stabilize UI deps for this test
jest.mock('@/components/ui/button', () => ({
  Button: ({ children, ...props }: any) => <button {...props}>{children}</button>
}));
jest.mock('lucide-react', () => ({
  ArrowLeft: (props: any) => <div data-testid="icon-arrow-left" {...props} />,
  ChevronRight: (props: any) => <div data-testid="icon-chevron-right" {...props} />,
  Brain: (props: any) => <div data-testid="icon-brain" {...props} />,
}));

function renderWithIntl(ui: React.ReactNode) {
  return render(
    <NextIntlClientProvider locale="en" messages={enMessages}>
      {ui}
    </NextIntlClientProvider>
  );
}

describe('DiaryHeader', () => {
  it('renders simplified desktop header', () => {
    const { container } = renderWithIntl(
      <DiaryHeader
        isMobile={false}
        isCBTActive={true}
        cbtCurrentStep={'emotions' as any}
        onBack={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders mobile header', () => {
    const { container } = renderWithIntl(
      <DiaryHeader
        isMobile={true}
        isCBTActive={true}
        cbtCurrentStep={'thoughts' as any}
        onBack={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
