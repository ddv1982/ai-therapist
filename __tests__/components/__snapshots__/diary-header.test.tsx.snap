// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`DiaryHeader renders mobile header 1`] = `
<div>
  <div
    class="border-b bg-card/70 backdrop-blur-md sticky top-0 z-10"
  >
    <div
      class="max-w-4xl mx-auto px-3 py-2"
    >
      <div
        class="flex items-center justify-between gap-2"
      >
        <button
          class="flex items-center gap-1 h-8 px-2"
          variant="ghost"
        >
          <div
            class="w-4 h-4"
            data-testid="icon-arrow-left"
          />
          <span
            class="text-sm"
          >
            Back
          </span>
        </button>
        <span
          class="text-xs text-primary"
        >
          Step 
          3
          /
          9
        </span>
      </div>
      <div
        class="w-full bg-muted rounded-full h-1 mt-2"
      >
        <div
          class="bg-primary h-1 rounded-full transition-all duration-500 ease-out"
          style="width: 33.33333333333333%;"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`DiaryHeader renders simplified desktop header 1`] = `
<div>
  <div
    class="border-b bg-card/50 backdrop-blur-md sticky top-0 z-10"
  >
    <div
      class="max-w-4xl mx-auto px-4 sm:px-6 py-4"
    >
      <div
        class="flex items-center justify-between"
      >
        <div
          class="flex items-center gap-4"
        >
          <button
            class="flex items-center gap-2 hover:bg-accent hover:text-accent-foreground"
            variant="ghost"
          >
            <div
              class="w-4 h-4"
              data-testid="icon-arrow-left"
            />
            <span
              class="hidden sm:inline"
            >
              Back to Chat
            </span>
          </button>
          <div
            class="flex items-center gap-3"
          >
            <div
              class="h-12 w-12 rounded-xl bg-primary flex items-center justify-center shadow-lg"
            >
              <div
                class="w-6 h-6 text-primary-foreground"
                data-testid="icon-brain"
              />
            </div>
            <h1
              class="text-3xl font-semibold"
            >
              Interactive CBT Session
            </h1>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
